### ISSUE TITLE
FireTray for GTK3

### ISSUE THREAD

**Issue by Temlakos** [foudfou/FireTray#182]
Firefox cannot run with FireTray 0.5.6 enabled. It crashes on startup every time. I had to start Firefox in safe mode and disable FireTray to get Firefox to run at all.

This could relate to the other issue I created, namely that FireTray's system tray icons are not suitable for use in KDE Plasma 5 and simply do not appear. I had to use a third-party tray even to see the FireTray icon for Thunderbird. (Thunderbird does not crash with FireTray in place.)

The Fedora Project just released F22. Its new KDE spin jumped to Plasma 5 from KDE 4. That's why I am only now noticing this issue. But you will likely see a lot of reports on this as a lot of unhappy campers try to start Firefox in the new environment only to find they cannot.

**Comment by foudfou** [comment]
Thx for reporting. I could reproduce the problem in a fresh F22 virtual machine.
The relevant part is:
```
(firefox:23724): GLib-GObject-WARNING **: cannot register existing type 'GdkDisplay'
(firefox:23724): GLib-CRITICAL **: g_once_init_leave: assertion 'result != 0' failed
(firefox:23724): Gdk-CRITICAL **: IA__gdk_x11_display_get_xdisplay: assertion 'GDK_IS_DISPLAY (display)' failed
```
And the problem is `gdk_display_get_default()` is returning NULL. But I don't know why yet.

**Comment by Temlakos** [comment]
OK, so it wasn't my imagination. Nor my one machine, either. But I knew that: I saw this problem on two different machines, this after upgrading from F21 to F22.

As I said, FireTray 0.5.6 still runs with Thunderbird. And when I install "wmsystemtray," I get the same fully functioning system tray icon I'm used to seeing, that displays the number of unread messages. It's just that I have to see that in a tiny window off in a corner somewhere, and usually hidden, because I like to run Firefox and Thunderbird full-screen.

**Comment by DavidPrevot** [comment]
Hi,

I believe a similar issue has just been reported on Debian: https://bugs.debian.org/787803

“Some investigating with GDB shows GTK3 symbols in the stack being mixed with the GTK2 which [Thunderbird/Firefox] use.

**Comment by foudfou** [comment]
Thank you @DavidPrevot and @gcp !! It looks like I should use `libappindicator1` instead of `libappindicator3`. Wonder how it used to work previously... Will fix that ASAP.

**Comment by Temlakos** [comment]
Would that possibly solve the issue of FireTray's icons not playing nice with KDE's new Plasma 5 system tray?

**Comment by foudfou** [comment]
Not exactly: we need another lib for that, or libappindicator to implement missing features.

**Comment by foudfou** [comment]
Switching to `libappindicator1` doesn't help much. You can check the `indicator1` branch.
Obviously GTK3 libs are still loaded, **even with FireTray disabled**. This may be because of the GTK theme in KDE, if any. But I don't know how to change that.

**Comment by foudfou** [comment]
After some digging, it appears [Firefox 38.0 in Fedora22 is the GTK3 build](https://plus.google.com/+WorldofGnomeOrg/posts/fUVawfRX98p), already. And since GTK2 and GTK3 cannot be loaded together, obviously FireTray needs to be ported to GTK3.

I'm updating the subject of this thread to reflect the real problem.

**Comment by foudfou** [comment]
Intersting post about [Firefox, GTK3 and plugins](http://emilio.pozuelo.org/posts/75).

**Comment by HanFox** [comment]
Thunderbird Earlybird (Aurora) 42.0a2 now also crashes when FireTray is loaded.

**Comment by foudfou** [comment]
Gtk3 support added. See the 4d9816ee439dab1 and the `gtk3` branch.

**Comment by HanFox** [comment]
Nice. The gtk3 branch seems to be working well in Earlybird. Thanks for the great addon :+1:

**Comment by inguin** [comment]
GTK3 version does not work on my system: No tray icon appears, and the Preferences dialog lacks the Mail and Chat tabs. A build from the master branch works flawlessly.

System details:
 - Fedora 21, x86-64
 - KDE 4.14.9
 - Thunderbird 38.1.0

**Comment by HanFox** [comment]
@inguin: do you have libappindicator-gtk3 installed?

**Comment by inguin** [comment]
Yes, it's installed. It wasn't installed when I first built the add-on, though, so there is a chance that something wasn't properly updated.

**Comment by inguin** [comment]
Found it: The Makefile must be updated to include the new directories ctypes/linux/gtk[23] in modules_sources.

**Comment by foudfou** [comment]
@inguin my bad! Fixed in 96c0862. Thank you.
